modAionImpl/src/org/aion/zero/impl/pendingState/AionPendingStateImpl.java

```diff
@@ -429,12 +398,59 @@ else if (bestRepoNonce(txFrom).compareTo(txNonce) < 1) {
         return txResponses;
     }
 
-    private boolean inPool(BigInteger txNonce, AionAddress from) {
-        return (this.txPool.bestPoolNonce(from).compareTo(txNonce) > -1);
+    private void addPendingTxToBackupDatabase(AionTransaction tx) {
```

Comment:
Could the backup functionality be separated from the pending state? For example, an option would be for all transactions to pass through a backup class before arriving to the pending state. Validations can be omitted for the backup case and deletions are not necessary until the full dataset is read from disk again.

Reply:
Create Jira Ticket AKI-661.